### PR TITLE
Upgrade code to php 8.1

### PR DESCRIPTION
**Pull request by ColonelMoutarde**
Changes proposed in this pull request:

- Upgrade code to php 8.1

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [x] documentation updated

### CODE REVIEW

**Comment by Alkarex** [comment]
Looks generally good. 
https://github.com/FreshRSS/FreshRSS/pull/6711 will be merged first, though, so you may want to target that branch

**Comment by ColonelMoutarde** [comment]
> Looks generally good. #6711 will be merged first, though, so you may want to target that branch

I don't know how :(

**Comment by Alkarex** [comment]
> I don't know how :(

Otherwise don't worry, #6711 will be merged shortly after the release of 1.24.3 (which is also soon), so it should just work, maybe with a few conflicts, though

**Comment by Alkarex** [comment]
https://github.com/FreshRSS/FreshRSS/pull/6706 has just been merged

File: app/Controllers/authController.php [hunk 1]
```diff
@@ -72,27 +72,18 @@ public function loginAction(): void {
...
 						' ; Remote IP address=' . connectionRemoteAddress() . ']'
 					]
-				], false);
-				break;
-			case 'none':
-				// It should not happen!
-				Minz_Error::error(404);
-				break;
-			default:
-				// TODO load plugin instead
```
Comment (Alkarex) [inline app/Controllers/authController.php:93]: Please keep the comments

File: app/Controllers/feedController.php [hunk 2]
```diff
@@ -640,7 +640,7 @@ public static function actualizeFeeds(?int $feed_id = null, ?string $feed_url =
 			}
 			unset($entries);
 
-			if (rand(0, 30) === 1) {	// Remove old entries once in 30.
+			if (random_int(0, 30) === 1) {	// Remove old entries once in 30.
```
Comment (Alkarex) [inline app/Controllers/feedController.php]: Please keep `rand()` here. No need of the `random_int` overhead

File: app/Controllers/configureController.php [hunk 3]
```diff
@@ -270,7 +270,7 @@ public function archivingAction(): void {
 			];
 		if (!empty(FreshRSS_Context::userConf()->archiving['keep_period'])) {
 			$keepPeriod = FreshRSS_Context::userConf()->archiving['keep_period'];
-			if (preg_match('/^PT?(?P<count>\d+)[YMWDH]$/', $keepPeriod, $matches)) {
+			if (preg_match('/^PT?(?P<count>\d+)[YMWDH]$/', (string) $keepPeriod, $matches)) {
```
Comment (Alkarex) [inline app/Controllers/configureController.php]: In the different places where you have used some manual type casting, please try to see whether it could be typed higher up, to avoid the need of manual casting.
And if you do need a type cast, please without space `(string)$s`

File: app/Exceptions/BadUrlException.php [hunk 4]
```diff
@@ -3,7 +3,4 @@
 
 class FreshRSS_BadUrl_Exception extends FreshRSS_Feed_Exception {
 
-	public function __construct(string $url) {
-		parent::__construct('`' . $url . '` is not a valid URL');
```
Comment (Alkarex) [inline app/Exceptions/BadUrlException.php:7]: Why removed?

File: app/Models/Entry.php [hunk 5]
```diff
@@ -241,20 +241,20 @@ public function content(bool $withEnclosures = true, bool $allowDuplicateEnclosu
 
 			if (self::enclosureIsImage($enclosure)) {
 				$content .= '<p class="enclosure-content"><img src="' . $elink . '" alt="" title="' . $etitle . '" /></p>';
-			} elseif ($medium === 'audio' || strpos($mime, 'audio') === 0) {
+			} elseif ($medium === 'audio' || str_starts_with((string) $mime, 'audio')) {
```
Comment (Alkarex) [inline app/Models/Entry.php]: Example of manual casting, which should really be avoided. Should be fixed higher up.

File: app/Models/FormAuth.php [hunk 6]
```diff
@@ -70,7 +70,7 @@ public static function deleteCookie(): void {
 			@unlink(DATA_PATH . '/tokens/' . $token . '.txt');
 		}
 
-		if (rand(0, 10) === 1) {
+		if (random_int(0, 10) === 1) {
```
Comment (Alkarex) [inline app/Models/FormAuth.php]: Same here, no need of the overhead of a cryptographically secure generator

File: app/Models/Share.php [hunk 7]
```diff
@@ -108,15 +95,12 @@ public static function get(string $type): ?FreshRSS_Share {
...
-		$this->url_transform = $url_transform;
-		$this->help_url = $help_url;
-		$this->HTMLtag = $HTMLtag;
-		$this->isDeprecated = $isDeprecated;
-		$this->transforms = $transforms;
+	private function __construct(private readonly string $type, private readonly string $url_transform, private array $transforms, string $form_type,
+		private readonly string $help_url, string $method, private ?string $field, /**
+   * @phpstan-var 'button'|null
+   */
+	private readonly ?string $HTMLtag, private readonly bool $isDeprecated = false) {
```
Comment (Alkarex) [inline app/Models/Share.php]: Looks wrong

File: app/Models/TagDAO.php [hunk 8]
```diff
@@ -92,8 +92,8 @@ public function updateTagAttributes(int $id, array $attributes): int|false {
 	}
 
 	/**
-	 * @param non-empty-string $key
-	 */
+  * @param non-empty-string $key
```
Comment (Alkarex) [inline app/Models/TagDAO.php]: Wrong whitespace changes?

File: app/SQL/install.sql.mysql.php [hunk 9]
```diff
@@ -113,7 +113,7 @@
 DROP TABLE IF EXISTS `_entrytag`, `_tag`, `_entrytmp`, `_entry`, `_feed`, `_category`;
 SQL;
 
-$GLOBALS['SQL_UPDATE_MINOR'] = <<<'SQL'
+$GLOBALS['SQL_UPDATE_MINOR'] = <<<'SQL_WRAP'
```
Comment (Alkarex) [inline app/SQL/install.sql.mysql.php]: Why? Please revert

File: app/Utils/dotNotationUtil.php [hunk 10]
```diff
@@ -5,14 +5,14 @@ final class FreshRSS_dotNotation_Util
...
 	/**
-	 * Get an item from an array using "dot" notation.
-	 * Functions adapted from https://stackoverflow.com/a/39118759
-	 * https://github.com/illuminate/support/blob/52e8f314b8043860b1c09e5c2c7e8cca94aafc7d/Arr.php#L270-L305
-	 * Newer version in
-	 * https://github.com/laravel/framework/blob/10.x/src/Illuminate/Collections/Arr.php#L302-L337
-	 *
-	 * @param \ArrayAccess<string,mixed>|array<string,mixed>|mixed $array
-	 */
+  * Get an item from an array using "dot" notation.
```
Comment (Alkarex) [inline app/Utils/dotNotationUtil.php]: Wrong whitespace changes, and more instances further below

File: app/Exceptions/BadUrlException.php (hunk 4 above)
Comment (ColonelMoutarde) [reply app/Exceptions/BadUrlException.php:7]: Never used

File: app/Models/FormAuth.php (hunk 6 above)
Comment (ColonelMoutarde) [reply app/Models/FormAuth.php]: done

File: app/Exceptions/BadUrlException.php (hunk 4 above)
Comment (ColonelMoutarde) [reply app/Exceptions/BadUrlException.php:7]: never used, dead code

File: app/Controllers/configureController.php (hunk 3 above)
Comment (Alkarex) [reply app/Controllers/configureController.php]: I suggest you move the type casts to another PR, so this PR can remain clean and focused on syntax

File: app/Controllers/feedController.php [hunk 11]
```diff
@@ -640,7 +640,7 @@ public static function actualizeFeeds(?int $feed_id = null, ?string $feed_url =
 			}
 			unset($entries);
 
-			if (rand(0, 30) === 1) {	// Remove old entries once in 30.
+			if (mt_rand(0, 30) === 1) {	// Remove old entries once in 30.
```
Comment (Alkarex) [inline app/Controllers/feedController.php]: Please leave out those `rand` changes. They are unrelated to upgrading to a newer syntax.

File: app/Controllers/javascriptController.php [hunk 12]
```diff
@@ -75,7 +75,7 @@ public function nonceAction(): void {
 		$this->view->salt1 = sprintf('$2a$%02d$', FreshRSS_password_Util::BCRYPT_COST);
 		$alphabet = './ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
 		for ($i = 22; $i > 0; $i--) {
-			$this->view->salt1 .= $alphabet[random_int(0, 63)];
+			$this->view->salt1 .= $alphabet[mt_rand(0, 63)];
```
Comment (Alkarex) [inline app/Controllers/javascriptController.php]: This one was better with `random_int` as this one benefits from a heavier cryptographically secure RNG.
Please do not change them in this PR, as they are unrelated to the topic of the PR

File: app/Exceptions/BadUrlException.php (hunk 4 above)
Comment (Alkarex) [reply app/Exceptions/BadUrlException.php:7]: It is used. See
https://github.com/FreshRSS/FreshRSS/blob/93c102d7eab69fee76f50fe207f9d3f4022372ca/app/Models/Feed.php#L274-L275

Please this PR focussed on the syntax update, with as little as possible of unrelated changes

Comment (Frenzie) [reply app/Exceptions/BadUrlException.php:7]: As a bit of an aside and depending on the specifics, available for extensions can also equal unused.

**Comment by Alkarex** [comment]
What is the status here? There are still plenty of unrelated changes such as casts, changes of `random`, wrong whitespace changes, code wrongly removed, etc.
Please focus, so this PR can be completed and merged

**Comment by ColonelMoutarde** [comment]
> What is the status here? There are still plenty of unrelated changes such as casts, changes of `random`, wrong whitespace changes, code wrongly removed, etc. Please focus, so this PR can be completed and merged

All is done

File: app/Controllers/authController.php [hunk 13]
```diff
@@ -136,7 +128,7 @@ public function formLoginAction(): void {
 				return;
 			}
 
-			usleep(random_int(100, 10000));	//Primitive mitigation of timing attacks, in μs
+			usleep(rand(100, 10000));	//Primitive mitigation of timing attacks, in μs
```
Comment (Alkarex) [inline app/Controllers/authController.php]: Could you please keep this PR clean of unrelated modifications like changing the random functions?
If any is worth doing, let's address it in another PR

File: app/Controllers/configureController.php [hunk 14]
```diff
@@ -270,7 +270,7 @@ public function archivingAction(): void {
 			];
 		if (!empty(FreshRSS_Context::userConf()->archiving['keep_period'])) {
 			$keepPeriod = FreshRSS_Context::userConf()->archiving['keep_period'];
-			if (preg_match('/^PT?(?P<count>\d+)[YMWDH]$/', $keepPeriod, $matches)) {
+			if (preg_match('/^PT?(?P<count>\d+)[YMWDH]$/', (string)$keepPeriod, $matches)) {
```
Comment (Alkarex) [inline app/Controllers/configureController.php]: I would better like to address the typing aspects in another PR, as I believe there are better ways than manually casting to `(string)`, so please no change of this type in this PR

File: app/Models/BooleanSearch.php [hunk 15]
```diff
@@ -4,20 +4,17 @@
...
-	 * @phpstan-var 'AND'|'OR'|'AND NOT'|'OR NOT'
-	 */
-	private string $operator;
-
 	/** @param 'AND'|'OR'|'AND NOT'|'OR NOT' $operator */
-	public function __construct(string $input, int $level = 0, string $operator = 'AND', bool $allowUserQueries = true) {
-		$this->operator = $operator;
+	public function __construct(string $input, int $level = 0, /**
+  * @phpstan-var 'AND'|'OR'|'AND NOT'|'OR NOT'
+  */
```
Comment (Alkarex) [inline app/Models/BooleanSearch.php]: Wrong change of comment, wrong indenting / whitespace

File: app/Utils/dotNotationUtil.php [hunk 16]
```diff
@@ -5,14 +5,14 @@ final class FreshRSS_dotNotation_Util
...
-	 * @param \ArrayAccess<string,mixed>|array<string,mixed>|mixed $array
-	 */
+	* Get an item from an array using "dot" notation.
+	* Functions adapted from https://stackoverflow.com/a/39118759
+	* https://github.com/illuminate/support/blob/52e8f314b8043860b1c09e5c2c7e8cca94aafc7d/Arr.php#L270-L305
+	* Newer version in
+	* https://github.com/laravel/framework/blob/10.x/src/Illuminate/Collections/Arr.php#L302-L337
+	*
+	* @param \ArrayAccess<string,mixed>|array<string,mixed>|mixed $array
+	*/
```
Comment (Alkarex) [inline app/Utils/dotNotationUtil.php]: Wrong change of comment indenting and again, please do not perform unrelated edits

File: composer.lock
Comment (Alkarex) [inline composer.lock:1]: Please no change of `composer.lock` in this PR. This is completely unrelated

**Comment by Alkarex** [comment]
> > What is the status here? There are still plenty of unrelated changes such as casts, changes of `random`, wrong whitespace changes, code wrongly removed, etc. Please focus, so this PR can be completed and merged
> 
> All is done

There are still plenty of unrelated changes such as casts, changes of `random`, wrong whitespace changes. I have just made a few comments higher up to give examples of those different categories of problems.
And the tests are failing.

File: app/Controllers/authController.php [hunk 17]
```diff
@@ -136,7 +128,7 @@ public function formLoginAction(): void {
 				return;
 			}
 
-			usleep(random_int(100, 10000));	//Primitive mitigation of timing attacks, in μs
+			usleep(randdom_int(100, 10000));	//Primitive mitigation of timing attacks, in μs
```
Comment (Alkarex) [inline app/Controllers/authController.php]: Please no change. Now there is a typo

File: app/Controllers/importExportController.php [hunk 18]
```diff
@@ -243,7 +237,7 @@ private function ttrssXmlToJson(string $xml): string|false {
 			$item = array_filter($item, static fn($v) =>
 				// Filter out empty properties, potentially reported as empty objects
 				(is_string($v) && trim($v) !== '') || !empty($v));
-			$item['updated'] = isset($item['updated']) ? strtotime($item['updated']) : '';
+			$item['updated'] = isset($item['updated']) ? strtotime((string)$item['updated']) : '';
```
Comment (Alkarex) [inline app/Controllers/importExportController.php]: Still some `(string)` manual type casts, which I would better like to address in another PR and in a better way

File: app/Models/BooleanSearch.php [hunk 19]
```diff
@@ -4,20 +4,16 @@
...
-	 * @phpstan-var 'AND'|'OR'|'AND NOT'|'OR NOT'
-	 */
-	private string $operator;
-
 	/** @param 'AND'|'OR'|'AND NOT'|'OR NOT' $operator */
-	public function __construct(string $input, int $level = 0, string $operator = 'AND', bool $allowUserQueries = true) {
-		$this->operator = $operator;
+	public function __construct(string $input, int $level = 0, /**
+  	* @phpstan-var 'AND'|'OR'|'AND NOT'|'OR NOT' */
+	private readonly string $operator = 'AND', bool $allowUserQueries = true) {
```
Comment (Alkarex) [inline app/Models/BooleanSearch.php]: Still some wrong comments changes

File: p/api/fever.php [hunk 20]
```diff
@@ -69,7 +69,7 @@ final class FeverDAO extends Minz_ModelPdo
 	 */
 	private function bindParamArray(string $prefix, array $values, array &$bindArray): string {
 		$str = '';
-		for ($i = 0; $i < count($values); $i++) {
+		for ($i = 0, $iMax = count($values); $i < $iMax; $i++) {
```
Comment (Alkarex) [inline p/api/fever.php]: Please refrain from making unrelated changes, which complicate the review

**Comment by Alkarex** [comment]
Still plenty of manual type casts, unrelated changes, and tests failing

File: p/api/fever.php (hunk 20 above)
Comment (ColonelMoutarde) [reply p/api/fever.php]: [EA] 'count(...)' is used in a loop and is a low performing construction

Comment (ColonelMoutarde) [reply p/api/fever.php]: If you accept, i can move max before the loop

**Comment by ColonelMoutarde** [comment]
> Still plenty of manual type casts, unrelated changes, and tests failing

done for tests

File: p/api/fever.php (hunk 20 above)
Comment (Alkarex) [reply p/api/fever.php]: The problem is that those random changes are unrelated to the topic of the PR, which leads to lots of noise and inefficient PRs

File: app/Controllers/importExportController.php [hunk 21]
```diff
@@ -317,13 +311,13 @@ private function importJson(string $article_file, bool $starred = false): bool {
 			if (empty($item['origin'])) {
 				$item['origin'] = [];
 			}
-			if (empty($item['origin']['title']) || trim($item['origin']['title']) === '') {
+			if (empty($item['origin']['title']) || trim((string)$item['origin']['title']) === '') {
```
Comment (Alkarex) [inline app/Controllers/importExportController.php]: Still no

File: app/Controllers/javascriptController.php [hunk 22]
```diff
@@ -61,7 +61,7 @@ public function nonceAction(): void {
 				if (strlen($s) >= 60) {
 					//CRYPT_BLOWFISH Salt: "$2a$", a two digit cost parameter, "$", and 22 characters from the alphabet "./0-9A-Za-z".
 					$this->view->salt1 = substr($s, 0, 29);
-					$this->view->nonce = sha1($salt . uniqid('' . mt_rand(), true));
+					$this->view->nonce = sha1($salt . uniqid('' . rand(), true));
```
Comment (Alkarex) [inline app/Controllers/javascriptController.php]: And still no

File: p/api/fever.php (hunk 20 above)
Comment (ColonelMoutarde) [reply p/api/fever.php]: done

File: app/Controllers/importExportController.php (hunk 21 above)
Comment (ColonelMoutarde) [reply app/Controllers/importExportController.php]: done

File: app/Controllers/javascriptController.php (hunk 22 above)
Comment (ColonelMoutarde) [reply app/Controllers/javascriptController.php]: done

File: app/Controllers/importExportController.php [hunk 23]
```diff
@@ -255,11 +249,11 @@ private function ttrssXmlToJson(string $xml): string|false {
 				$item['categories'][] = 'user/-/state/com.google/broadcast';
 			}
 			if (!empty($item['label_cache'])) {
-				$labels_cache = json_decode($item['label_cache'], true);
+				$labels_cache = json_decode((string)$item['label_cache'], true);
 				if (is_array($labels_cache)) {
 					foreach ($labels_cache as $label_cache) {
 						if (!empty($label_cache[1])) {
-							$item['categories'][] = 'user/-/label/' . trim($label_cache[1]);
+							$item['categories'][] = 'user/-/label/' . trim((string)$label_cache[1]);
```
Comment (Alkarex) [inline app/Controllers/importExportController.php]: It is quite incredible, but there are still many manual casts in this PR, so still not OK.
Those casts are not only unrelated to the PR, but several are also wrong, as one cannot cast `mixed` to `string`.
I will try to implement https://github.com/FreshRSS/FreshRSS/issues/6989 soonish, which will increase the typing in a better way.

Comment (ColonelMoutarde) [reply app/Controllers/importExportController.php]: I'm going to delete all the casts I've added... 
For your information, trim() only accepts strings as the first parameter. 
The code above never checks that a string is present...
I delete them for this function. I can't guarantee that the code won't break if something else is entered as an argument.

Comment (Alkarex) [reply app/Controllers/importExportController.php]: Sure, but:
1. It is unrelated to this PR;
2. It is not the correct fix.

File: app/Models/Entry.php [hunk 24]
```diff
@@ -226,7 +226,7 @@ public function content(bool $withEnclosures = true, bool $allowDuplicateEnclosu
 			$description = nl2br($enclosure['description'] ?? '', true);
 			$length = $enclosure['length'] ?? 0;
 			$medium = $enclosure['medium'] ?? '';
-			$mime = $enclosure['type'] ?? '';
+			$mime = ($enclosure['type'] ?? '');
```
Comment (Alkarex) [inline app/Models/Entry.php]: Any reason for this change?

File: app/Models/Entry.php [hunk 25]
```diff
@@ -568,21 +568,12 @@ public function matches(FreshRSS_BooleanSearch $booleanSearch): bool {
...
-					case 'AND':
-					default:
-						$ok &= $this->matches($filter);
-						break;
-				}
+				match ($filter->operator()) {
+					'OR' => $ok |= $this->matches($filter),
+					'OR NOT' => $ok |= !$this->matches($filter),
+					'AND NOT' => $ok &= !$this->matches($filter),
+					default => $ok &= $this->matches($filter),
```
Comment (Alkarex) [inline app/Models/Entry.php:576]: Keep the information about `AND` explicitly

Comment (Alkarex) [inline app/Models/Entry.php]:
```suggestion
					'AND' => $ok &= $this->matches($filter),
					'OR' => $ok |= $this->matches($filter),
					'OR NOT' => $ok |= !$this->matches($filter),
					'AND NOT' => $ok &= !$this->matches($filter),
```

File: app/Models/Themes.php [hunk 26]
```diff
@@ -165,14 +165,10 @@ public static function icon(string $name, int $type = self::ICON_DEFAULT): strin
...
-			case self::ICON_IMG:
-				return '<img class="icon" src="' . Minz_Url::display($url) . '" loading="lazy" alt="' . $alt . '"' . $title . ' />';
-			case self::ICON_EMOJI:
-			default:
-				return '<span class="icon"' . $title . '>' . $alt . '</span>';
-		}
+		return match ($type) {
+			self::ICON_URL => Minz_Url::display($url),
+			self::ICON_IMG => '<img class="icon" src="' . Minz_Url::display($url) . '" loading="lazy" alt="' . $alt . '"' . $title . ' />',
+			default => '<span class="icon"' . $title . '>' . $alt . '</span>',
```
Comment (Alkarex) [inline app/Models/Themes.php:172]: Keep `self::ICON_EMOJI` explicitly

File: cli/do-install.php [hunk 27]
```diff
@@ -92,10 +92,10 @@ public function __construct() {
...
 ];
 
-$config = array(
-		'salt' => generateSalt(),
-		'db' => FreshRSS_Context::systemConf()->db,
-	);
+$config = [
+	'salt' => generateSalt(),
+	'db' => FreshRSS_Context::systemConf()->db,
+	];
```
Comment (Alkarex) [inline cli/do-install.php]:
```suggestion
];
```

File: cli/user-info.php [hunk 28]
```diff
@@ -84,7 +84,7 @@ public function __construct() {
 		'tags' => $tagDAO->count(),
 		'lang' => FreshRSS_Context::userConf()->language,
 		'mail_login' => FreshRSS_Context::userConf()->mail_login,
-	);
+		];
```
Comment (Alkarex) [inline cli/user-info.php]:
```suggestion
	];
```

File: tests/app/Models/SearchTest.php [hunk 29]
```diff
@@ -166,14 +168,14 @@ public static function test__construct_whenInputContainsPubdate_setsPubdateValue
...
-			array('pubdate:/2008-05-11', null, strtotime('2008-05-12') - 1),
-		);
+		return [
+			['pubdate:2007-03-01T13:00:00Z/2008-05-11T15:30:00Z', 1172754000, 1210519800],
+			['pubdate:2007-03-01T13:00:00Z/P1Y2M10DT2H30M', 1172754000, 1210519799],
+			['pubdate:P1Y2M10DT2H30M/2008-05-11T15:30:00Z', 1172754001, 1210519800],
+			['pubdate:2007-03-01/2008-05-11', strtotime('2007-03-01'), strtotime('2008-05-12') - 1],
+			['pubdate:2007-03-01/', strtotime('2007-03-01'), null],
+			['pubdate:/2008-05-11', null, strtotime('2008-05-12') - 1],
+			];
```
Comment (Alkarex) [inline tests/app/Models/SearchTest.php]:
```suggestion
		];
```

Plenty of wrongly indented arrays. Please check them all

File: tests/app/Models/SearchTest.php [hunk 30]
```diff
@@ -229,56 +231,51 @@ public static function test__construct_whenInputContainsMultipleKeywords_setsVal
...
-				strtotime('2007-03-01'),
-				strtotime('2008-05-12') - 1,
-				array('word4', 'word5'),
-				null,
-			),
-			array(
-				'word6 intitle:word2 inurl:word3 pubdate:2007-03-01/2008-05-11 #word4 author:word1 #word5 date:2007-03-01/2008-05-11',
-				array('word1'),
+		return [
+			['author:word1 date:2007-03-01/2008-05-11 intitle:word2 inurl:word3 pubdate:2007-03-01/2008-05-11 #word4 #word5',
```
Comment (Alkarex) [inline tests/app/Models/SearchTest.php]: Another example of wrong indenting

File: tests/app/Models/SearchTest.php (hunk 29 above)
Comment (Alkarex) [reply tests/app/Models/SearchTest.php]: There were too many errors in this file, so I just autoformatted the arrays https://github.com/FreshRSS/FreshRSS/pull/6748/commits/52c74d324b58d6762319b2ad6631879662eeca8e

File: app/Controllers/authController.php [hunk 31]
```diff
@@ -72,27 +72,19 @@ public function loginAction(): void {
...
+			'http_auth' => Minz_Error::error(403, [
 					'error' => [
 						_t('feedback.access.denied'),
 						' [HTTP Remote-User=' . htmlspecialchars(httpAuthUser(false), ENT_NOQUOTES, 'UTF-8') .
 						' ; Remote IP address=' . connectionRemoteAddress() . ']'
 					]
-				], false);
-				break;
-			case 'none':
-				// It should not happen!
```
Comment (Alkarex) [inline app/Controllers/authController.php:89]: Please keep that comment as well

File: p/api/greader.php [hunk 32]
```diff
@@ -266,44 +266,37 @@ private static function userInfo(): never {
...
 		foreach ($categories as $cat) {
-			$tags[] = array(
-				'id' => 'user/-/label/' . htmlspecialchars_decode($cat->name(), ENT_QUOTES),
-				//'sortid' => $cat->name(),
-				'type' => 'folder',	//Inoreader
-			);
+			$tags[] = [
+			'id' => 'user/-/label/' . htmlspecialchars_decode($cat->name(), ENT_QUOTES),
+			//'sortid' => $cat->name(),
+			'type' => 'folder',
```
Comment (Alkarex) [inline p/api/greader.php]: Several formatting issues in that file as well

File: p/api/greader.php [hunk 33]
```diff
@@ -354,14 +347,14 @@ private static function subscriptionList(): never {
 					],
 					//'sortid' => $feed->name(),
 					//'firstitemmsec' => 0,
-					'url' => htmlspecialchars_decode($feed->url(), ENT_QUOTES),
-					'htmlUrl' => htmlspecialchars_decode($feed->website(), ENT_QUOTES),
+					'url' => htmlspecialchars_decode((string)$feed->url(), ENT_QUOTES),
```
Comment (Alkarex) [inline p/api/greader.php]: And still some unrelated casts here

File: app/Models/Entry.php (hunk 24 above)
Comment (ColonelMoutarde) [reply app/Models/Entry.php]: Parenthesis is useless

Comment (Alkarex) [reply app/Models/Entry.php]: Indeed, so why did you **add** the parentheses?

Comment (ColonelMoutarde) [reply app/Models/Entry.php]: Oups ! Good review

File: p/api/greader.php [hunk 34]
```diff
@@ -489,25 +484,25 @@ private static function unreadCount(): never {
...
-				);
+					];
 				if ($catLastUpdate < $lastUpdate) {
 					$catLastUpdate = $lastUpdate;
 				}
 			}
-			$unreadcounts[] = array(
-				'id' => 'user/-/label/' . htmlspecialchars_decode($cat->name(), ENT_QUOTES),
+			$unreadcounts[] = [
+				'id' => 'user/-/label/' . htmlspecialchars_decode((string)$cat->name(), ENT_QUOTES),
```
Comment (Alkarex) [inline p/api/greader.php]: Still not good (and this is not the only example)

File: p/api/greader.php [hunk 35]
```diff
@@ -266,44 +266,37 @@ private static function userInfo(): never {
...
-				//'sortid' => $label->name(),
-				'type' => 'tag',	//Inoreader
-				'unread_count' => $label->nbUnread(),	//Inoreader
-			);
+			$tags[] = [
+			'id' => 'user/-/label/' . htmlspecialchars_decode((string)$label->name(), ENT_QUOTES),
+			//'sortid' => $label->name(),
+			'type' => 'tag',
+			//Inoreader
+			'unread_count' => $label->nbUnread(),
```
Comment (Alkarex) [inline p/api/greader.php]: Still several examples of wrong array indenting

File: p/api/greader.php [hunk 36]
```diff
@@ -656,21 +647,12 @@ private static function streamContents(string $path, string $include_target, int
...
-			case 'starred':
-				$type = 's';
-				break;
-			case 'feed':
-				$type = 'f';
-				break;
-			case 'label':
-				$type = 'c';
-				break;
-			case 'reading-list':
```
Comment (Alkarex) [inline p/api/greader.php:675]: Here as well, keep `'reading-list'` explicitly

Comment (Alkarex) [reply p/api/greader.php:675]: https://github.com/FreshRSS/FreshRSS/pull/6748/commits/dbfdecf9c133e63dfbc7ff83189ade9f7c0abfbe

**Review by Alkarex — APPROVED** [review]
